packages/nuxeo-retention-package/src/main/resources/install/templates/retention/nxserver/config/retention-compliance-config.xml.nxftl

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0"?>
+
+<component name="org.nuxeo.retention.compliance.types">
+
+    <require>org.nuxeo.retention.types</require>
```

Comment:
Why not in the `if`? If the mode is not `compliance`, there is no need for this component to require something as it will be empty.

Reply:
My rational here was to avoid this contribution to be loaded before `retention.types` since I wasn't able to understand how this engine works and if there is any chance of that to happen.

Reply:
It is not possible to have that scenario? If not I'll move that line 🙂

Reply:
You're right, it's better to have the `require`, but it's only needed when the component is not empty, so you can move that require with the other one, inside the `if`. If the component is empty, we don't care if it's loaded before or after `org.nuxeo.retention.types`.